# Capstone Project - Car Accident Severity (Week 2)
__Applied Data Science Capstone By IBM/Coursera__

## Introduction: Business Problem

The project is to predict the severity of the Accidents in Seattle in certain areas and their causes. By observations through the given dataset containing information and stats about the types and causes of the accidents, I have to predict the severity of the accident depending upon traffic delay, cars or buses involved and etc which would cause or play important role in prediction. 

In the given dataset, the data is provided like overspeeding, number of cars, buses, weather and road conditions. For prediction of severity we apply machine learning on dependent variable i.e severity. Therefore, Supervised learning will be used for this prediction.

# Data

By analyzing the data we have certain insights from the data which is provided:
1. Types of injuries
2. Location of Accidents
3. Weather/Road/Driver's Conditions
4. Cars/buses/pedestrians involved and their injury severity.

To help in predicting severity of accidents, we create dependent and independent variables and then apply train_test_split to split the data so that it can be applied to machine learning alogrithm which we will apply. We have to predict severity so it will be our dependent variable.

Machine Learning algorithm should be chosen wisely to predict more accurately and give less complicated results i.e results which are easily understandable.

In [1]:
import pandas as pd
import numpy as np

In [3]:
# The code was removed by Watson Studio for sharing.

/opt/conda/envs/Python36/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3020: DtypeWarning: Columns (33) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [4]:
df.head()

SEVERITYCODE           X          Y  OBJECTID  INCKEY  COLDETKEY REPORTNO  \
0             2 -122.323148  47.703140         1    1307       1307  3502005   
1             1 -122.347294  47.647172         2   52200      52200  2607959   
2             1 -122.334540  47.607871         3   26700      26700  1482393   
3             1 -122.334803  47.604803         4    1144       1144  3503937   
4             2 -122.306426  47.545739         5   17700      17700  1807429   

    STATUS      ADDRTYPE   INTKEY  ... ROADCOND                LIGHTCOND  \
0  Matched  Intersection  37475.0  ...      Wet                 Daylight   
1  Matched         Block      NaN  ...      Wet  Dark - Street Lights On   
2  Matched         Block      NaN  ...      Dry                 Daylight   
3  Matched         Block      NaN  ...      Dry                 Daylight   
4  Matched  Intersection  34387.0  ...      Wet                 Daylight   

  PEDROWNOTGRNT  SDOTCOLNUM SPEEDING ST_COLCODE  \
0           NaN         NaN      NaN         10   
1           NaN   6354039.0      NaN         11   
2           NaN   4323031.0      NaN         32   
3           NaN         NaN      NaN         23   
4           NaN   4028032.0      NaN         10   

                                          ST_COLDESC  SEGLANEKEY  \
0                                  Entering at angle           0   
1  From same direction - both going straight - bo...           0   
2                             One parked--one moving           0   
3                   From same direction - all others           0   
4                                  Entering at angle           0   

   CROSSWALKKEY  HITPARKEDCAR  
0             0             N  
1             0             N  
2             0             N  
3             0             N  
4             0             N  

[5 rows x 38 columns]

In [5]:
df.dtypes

SEVERITYCODE        int64
X                 float64
Y                 float64
OBJECTID            int64
INCKEY              int64
COLDETKEY           int64
REPORTNO           object
STATUS             object
ADDRTYPE           object
INTKEY            float64
LOCATION           object
EXCEPTRSNCODE      object
EXCEPTRSNDESC      object
SEVERITYCODE.1      int64
SEVERITYDESC       object
COLLISIONTYPE      object
PERSONCOUNT         int64
PEDCOUNT            int64
PEDCYLCOUNT         int64
VEHCOUNT            int64
INCDATE            object
INCDTTM            object
JUNCTIONTYPE       object
SDOT_COLCODE        int64
SDOT_COLDESC       object
INATTENTIONIND     object
UNDERINFL          object
WEATHER            object
ROADCOND           object
LIGHTCOND          object
PEDROWNOTGRNT      object
SDOTCOLNUM        float64
SPEEDING           object
ST_COLCODE         object
ST_COLDESC         object
SEGLANEKEY          int64
CROSSWALKKEY        int64
HITPARKEDCAR       object
dtype: objec

__Data is Splitted into Independent and Dependent variables.__

In [18]:
y = df["SEVERITYCODE"]
X = df[["X","Y","HITPARKEDCAR","ST_COLDESC","ST_COLCODE","SPEEDING","PEDROWNOTGRNT","LIGHTCOND","ROADCOND","WEATHER","UNDERINFL","INATTENTIONIND","JUNCTIONTYPE","VEHCOUNT","PEDCYLCOUNT","PEDCOUNT","PERSONCOUNT","ADDRTYPE"]]

In [20]:
X.shape

(194673, 18)

__By above shape method, we came to know that there are 194673 accidents that took place which is a good amount of data to predict our dependent variable.__